packages/calypso-products/src/plans-utilities.ts

```diff
@@ -41,10 +41,33 @@ export function getTermDuration( term: string ): number | undefined {
 
 export const redirectCheckoutToWpAdmin = (): boolean => !! JETPACK_REDIRECT_CHECKOUT_TO_WPADMIN;
 
-export const is2023PricingGridEnabled = () => {
-	const supportedLocales = [ 'en', 'en-gb', 'es' ];
-	return (
-		isEnabled( 'onboarding/2023-pricing-grid' ) &&
-		supportedLocales.includes( getLocaleSlug() ?? '' )
-	);
+export const is2023PricingGridEnabled = (): boolean => {
+	const isFeatureFlagEnabled = (): boolean => isEnabled( 'onboarding/2023-pricing-grid' );
+
+	const isLocaleSupported = (): boolean => {
+		const supportedLocales = [ 'en', 'en-gb', 'es' ];
+		return supportedLocales.includes( getLocaleSlug() ?? '' );
+	};
+
+	const isActivePage = (): boolean => {
+		const currentRoutePath = window.location.pathname;
+
+		// Is this the internal plans page /plans/<site-slug> ?
+		if (
+			currentRoutePath.includes( 'plans' ) &&
+			! currentRoutePath.includes( 'start' ) &&
+			! currentRoutePath.includes( 'setup' )
+		) {
+			return true;
+		}
```

Comment:
If I'm reading this correctly, it assumes that a site slug cannot contain the words `start` or `setup`. It's a fairly ambiguous match using `includes` this way.

Can't a simple regex match work? simple as `/plans\/.+/g`. do we need the other exclusions?

Reply:
`/plans\/.+/g` will return match true for `/start/plans` right? I want to avoid this. Is it possible to avoid `start` or `setup` but match for any other string to match a siteSlug?

Maybe i should split path parts and then check them to be sure 🤔

Reply:
> but is it necessary? 

Yes the initial change was decided last Thursday to only be shipped to `/plans` and `/start/plans` since there are several flows that use this component and we want to reduce the area this change effects. Also many other pages were showing sketchy/broken UI similar to what happened in the add domain flow.

> Please read up on regex. try https://regex101.com/

I can handle most basic regex but need to read up on more advance uses

> You can mark the beginning of the string in your regex e.g.

So these are a few test cases to match in this specific instance

`'/plans'` - should match
`'/start/hosting/plans'` - should not match
`'/plans/yourgroovydomain.com` - should match
`'/plans/myhosting.com` - should match

I added a more straightforward unambigous way to arrive at the correct paths 

https://github.com/Automattic/wp-calypso/pull/73481/commits/8c167ad57b82ad47061e93fc3cab0796020cd121

Can you have another look :)

Reply:
> I just realised what you said and you are correct, that should be enough 💯

Well, yes and no. `'/plans/123'.match(/^\/plans\/.+/g)` wouldn't match `/plans`. The cases you outline are quite simple, so there isn't a need for anything overly-engineered here, like manual string splitting and things like that. I'd still go for regex as anything regex-based can be easily adjusted to anything that may arise - but maybe just native string matching (like `startsWith` mentioned below) are more than sufficient (and unambiguous) for this case. Also potentially more performant as they probably won't traverse the whole string.

> Yes the initial change was decided last Thursday to only be shipped to /plans and /start/plans since there are several flows that use this component and we want to reduce the area this change effects. Also many other pages were showing sketchy/broken UI similar to what happened in the add domain flow.

I wonder if any caveats with the approach (to restrict the truthness of `is2023PricingGridEnabled` to only when the user is on one of these pages). There is no constrained usage of that method i.e. it can be called from anywhere in the app. Any chance that this may lead to ambiguity in the environment, in general?

Reply:
This is a good observation @chriskmnds, let me go back to the drawing board with this.